crates/ghost_actor/src/ghost_actor.rs

```diff
@@ -436,14 +465,13 @@ mod tests {
 
             // In this test actor we simply take all the messages we get and
             // add them to our internal state.
-            for mut msg in self.endpoint_as_child.as_mut().drain_messages() {
-                let payload = match msg.take_message().expect("exists") {
+            for (payload, responder) in self.endpoint_as_child.as_mut().drain_messages() {
+                let payload = match payload {
                     TestMsgIn(payload) => payload,
                 };
                 self.internal_state.push(payload.clone());
-                if msg.is_request() {
-                    msg.respond(Ok(TestMsgInResponse(format!("we got: {}", payload))))
-                };
+                responder(
```

Comment:
I don't like how the possibility of messages being events not requests becomes obscured here...